docs/fundamentals/code-analysis/quality-rules/ca1845.md

```diff
@@ -0,0 +1,66 @@
+---
+title: "CA1845: Use span-based 'string.Concat' (analysis rule)"
+description: "Learn about code analysis rule CA1845: Use span-based 'string.Concat'"
+ms.date: 05/18/2021
+ms.topic: reference
+f1_keywords:
+  - "UseSpanBasedStringConcat"
+  - "CA1845"
+helpviewer_keywords:
+  - "UseSpanBasedStringConcat"
+  - "CA1845"
+author: NewellClark
+dev_langs:
+  - CSharp
+  - VB
+---
+# CA1845: Use span-based 'string.Concat'
+
+| | Value |
+|-|-|
+| **Rule ID** |CA1845|
+| **Category** |[Performance](performance-warnings.md)|
+| **Fix is breaking or non-breaking** |Non-breaking|
+
+## Cause
+
+This rule locates string-concatenation expressions that contain `Substring` calls and suggests replacing `Substring` with `AsSpan` and using the span-based overload of `string.Concat`.
+
+## Rule description
+
+Calling `Substring` produces a copy of the extracted substring. By using `AsSpan` instead of `Substring` and calling the overload of `string.Concat` which accepts spans, we can eliminate this unnecessary string allocation.
```

Comment:
```suggestion
Calling `Substring` produces a copy of the extracted substring. By using `AsSpan` instead of `Substring` and calling the overload of `string.Concat` that accepts spans, you can eliminate the unnecessary string allocation.
```